conda/models/environment.py

```diff
@@ -8,19 +8,91 @@
 from logging import getLogger
 from typing import TYPE_CHECKING
 
-from ..base.constants import PLATFORMS
+from ..base.constants import (
+    PLATFORMS,
+)
 from ..exceptions import CondaValueError
 
 if TYPE_CHECKING:
-    from typing import Any
-
+    from ..base.constants import (
+        ChannelPriority,
+        DepsModifier,
+        SatSolverChoice,
+        UpdateModifier,
+    )
     from .match_spec import MatchSpec
     from .records import PackageRecord
 
 
 log = getLogger(__name__)
 
 
+@dataclass
+class EnvironmentConfig:
+    """
+    **Experimental** While experimental, expect both major and minor changes across minor releases.
+
+    Data model for a conda environment config.
+    """
+
+    aggressive_update_packages: bool | None = None
+
+    channel_priority: ChannelPriority | None = None
+
+    channels: list[str] = field(default_factory=list)
+
+    channel_settings: dict[str, str] = field(default_factory=dict)
+
+    deps_modifier: DepsModifier | None = None
+
+    disallowed_packages: list[str] = field(default_factory=list)
+
+    pinned_packages: list[str] = field(default_factory=list)
+
+    repodata_fns: list[str] = field(default_factory=list)
+
+    solver: SatSolverChoice | None = None
```

Comment:
Is this `SatSolverChoice`? I think this is incorrect; this is supposed to be the solver plugin, which cannot be known statically, so you'll need to set this to `str`.
```suggestion
    solver: str | None = None
```

Reply:
ah ya `sat_solver` and `solver` are 2 different settings and both should be represented, like 
```
sat_solver: SatSolverChoice | None = None
solver: str | None = None
```